server/api/runtime_handlers/daskjob.py

```diff
@@ -320,8 +320,29 @@ def enrich_dask_cluster(
         # TODO: we might never enter here, since running a function requires defining an image
         or "daskdev/dask:latest"
     )
-    env = spec.env
-    env.extend(function.generate_runtime_k8s_env())
+    env = function.generate_runtime_k8s_env()
+
+    # filter any spec.env that already exists in env
+    # in other words, dont let spec.env override env (or not even duplicate it)
+    # we dont want to override env to ensure k8s runtime envs are enforced and correct
+    # leaving no room for human mistakes
+    env.extend(
+        filter(
+            lambda spec_env: isinstance(spec_env, dict)
+            and not any(
```

Comment:
You are excluding all env vars that are `V1EnvVar` with this `isinstance(spec_env, dict) and ...`. Looks like a mistake

Reply:
ahh right